File: Go/Soft.go
package main
import "fmt"

type Soft struct {
	Tyres
}

func NewSoft(A, B, C, D string, lapsMax int, isWet bool, tyreColor string) Soft {
	tyres := NewTyres(A, B, C, D, lapsMax, isWet, tyreColor)
	return Soft{Tyres: tyres}
}
 
func (s Soft) Durability(laps int) string {
	if laps > s.durabilityMax {
		s.Pits()
		return fmt.Sprintf("The tyres have been on track %d laps\nThe Softs are being changed", laps)
	}
	return fmt.Sprintf("The tyres have been on track %d laps\nCan continue", laps)
}

func (s Soft) PSI() {
	fmt.Println("The air pressure required is 34")
}

File: JavaScript/Soft.js
import Tyres from "./Tyres.js";

// Esta clase es para las llantas blandas (Soft)
export default class Soft extends Tyres {
  // Constructor que delega en Tyres
  constructor(A, B, C, D, lapsMax, isWet, tyreColor) {
    super(A, B, C, D, lapsMax, isWet, tyreColor);
  }

  // Si supera la durabilidad máxima, entra a pits
  Durability(laps) {
    if (laps > this.durabilityMax) {
      this.Pits();
      return (
        `the tyres have been on track ${laps} laps\nThe Softs are being changed`
      );
    }
    return `the tyres have been on track ${laps} laps\nCan continue`;
  }

  // Presión recomendada para Soft
  PSI() {
    console.log("The air pressure required is 34");
  }
}